### ISSUE TITLE
BelongsToMany dropdown not populating after editing relationship

### ISSUE THREAD

**Issue by jamespavett** [laravel/nova-issues#3368]
- Laravel Version: 8.40.0
- Nova Version: 3.24.0
- PHP Version: 7.4.3
- Database Driver & Version: 8.0.22
- Operating System and Version: Ubuntu 16.04
- Browser type and version: Google Chrome

### Description:

I have a BelongsToMany relationship shown within a resource, which acts on a pivot table with a couple of custom columns.

When modifying the relationship, as expected, the second field is an uneditable dropdown containing a list of the linked models (Called Individual Site Queries in below screenshot).

![image](https://user-images.githubusercontent.com/10764464/116579392-41954200-a90a-11eb-8d8b-5bdc519a7a09.png)

However when clicking on the _Update & Continue Editing_ option, the page reloads but the dropdown is no longer populated with a value, instead it is showing the option to Choose a new option. If I use F5 to refresh the page the value gets populated as expected though.

![image](https://user-images.githubusercontent.com/10764464/116579591-6ee1f000-a90a-11eb-963d-50627127bec4.png)

![image](https://user-images.githubusercontent.com/10764464/116579465-5245b800-a90a-11eb-858f-e35840019217.png)

I can still update / modify the relationship as normal, but it is an annoying visual bug for the end user.

### Detailed steps to reproduce the issue on a fresh Nova installation:

In this instance I have a Many to Many relationship between two models, with them being links in Nova as follows:

```
BelongsToMany::make('Individual Site Queries', 'sites', SiteResource::class)->fields(function () {
                return [
                    Text::make('Version', 'version')->sortable()->readonly(),
                    Boolean::make('Template Syntax', 'template_syntax')->sortable(),
                ];
            }),
```

**Comment by crynobone** [comment]
This is expected behaviour.marking this as feature request

**Comment by davidhemphill** [comment]
👋 Hi there! In order to keep this repository focused on bug reports, we auto-close feature requests and requests for help. Feel free to post your feature requests so others can discuss and add reactions. We'll keep an eye on them for later planning.

**Comment by jamespavett** [comment]
@davidhemphill is this not being considered a bug / defect? I can see that the label was assigned earlier, but then the issue has just been closed.

**Comment by crynobone** [comment]
Closed assigned bug report only meant a PR has been merge and pending for next patch release .

**Comment by jamespavett** [comment]
Okay cheers thanks for getting back to me

**Comment by crynobone** [comment]
This has been fixed with [v3.25.0 release](https://nova.laravel.com/releases/3.25.0).
